zebrad/src/config.rs

```diff
@@ -15,11 +15,11 @@ use zebra_network::Config as NetworkSection;
 #[serde(deny_unknown_fields)]
 pub struct ZebradConfig {
     /// Tracing configuration
-    pub tracing: TracingSection,
+    pub tracing: Option<TracingSection>,
     /// Networking configuration
-    pub network: NetworkSection,
+    pub network: Option<NetworkSection>,
     /// Metrics configuration
-    pub metrics: MetricsSection,
+    pub metrics: Option<MetricsSection>,
```

Comment:
Making these `Option`s is motivated by the empty-config problem, right? I don't think they should be optional.  What is someone supposed to do if the field is missing?

Reply:
Yea, that and the fact that if nothing is optional I cannot tell if the `config.tracing.filter` value was provided by an actual config file or just the default settings. I think the latter issue means these have to be optional, but I think theres still an issue where if you have the header for one of these sections present then all the values become required again.

I don't think this is the right solution but I think this is strictly better than the current situation. What do you think about merging this and opening an issue to rework all the config stuff to support all partial configs.

Reply:
Why do we need to check whether the `filter` value is a default setting? So that we know whether it should take priority over an environment variable?  Did we decide whether we wanted env variables to override it or vice versa?

If the config variable is supposed to override the env variables, then I think we should make `config.tracing.filter` optional, so that it has semantic meaning (`Some` is an explicit preference, `None` is use the env variable).  My preference would be to not merge this, since I think it will force a lot of code to have to deal with fallibilty that shouldn't have to.

Reply:
The env variable part is pretty easy to resolve but I figured we should also have the `--verbose` cli flag take priority over it and theres no way to reference that in the `Default` impl for TracingSection